cmd/limactl/prune.go

```diff
@@ -17,15 +22,115 @@ func newPruneCommand() *cobra.Command {
 		ValidArgsFunction: cobra.NoFileCompletions,
 		GroupID:           advancedCommand,
 	}
+	pruneCommand.Flags().Bool("keep-referred", false, "Keep objects that are referred by some instances or templates")
 	return pruneCommand
 }
 
-func pruneAction(_ *cobra.Command, _ []string) error {
+func pruneAction(cmd *cobra.Command, _ []string) error {
+	keepReferred, err := cmd.Flags().GetBool("keep-referred")
+	if err != nil {
+		return err
+	}
 	ucd, err := os.UserCacheDir()
 	if err != nil {
 		return err
 	}
 	cacheDir := filepath.Join(ucd, "lima")
 	logrus.Infof("Pruning %q", cacheDir)
-	return os.RemoveAll(cacheDir)
+	if !keepReferred {
+		return os.RemoveAll(cacheDir)
+	}
+
+	// Prune downloads that are not used by any instances or templates
+	downloadDir := filepath.Join(cacheDir, "download", "by-url-sha256")
```

Comment:
This internal data structure should be handled in `pkg/downloader` rather than in the CLI

Reply:
Moved some codes to `pkg/downloader`.